### DISCUSSION TITLE
Isolate clients from each other

### DISCUSSION THREAD

**Discussion by azat385** [wg-easy/wg-easy#512] [Q&A]
How to isolate clients from each others?
in default configuration any client can reach to any client.

**Comment by Pleonazm** [comment]
You can add some iptables rules(like a basic example below). Problem is that you have to do it separately for each client, and of course it depends what other rules you have
On the client's side use -o, on the server's -i to designate the interface.

iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d WG_SERVER_IP -j ACCEPT
iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d THIS_PARTICULAR_IP_ALLOWED_FOR_CLIENT -j ACCEPT
iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d WG_SUBNET -j DROP

In the above example, a client can reach some selected IPs, a WG server, and nothing else on WG subnet (including other clients' WG interface). Note: if there is something behind other clients, then you have to block this IP also, so the client won't bypass the filter (iptables use data from the IP header)

**Reply by azat385** [reply]
> iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d WG_SERVER_IP -j ACCEPT
> iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d THIS_PARTICULAR_IP_ALLOWED_FOR_CLIENT -j ACCEPT
> iptables -A FORWARD -i WG_INT_NAME -s CLIENT_IP/32 -d WG_SUBNET -j DROP

allow connection to WG_SERVER:
`iptables -A FORWARD -i wg0 -s 10.8.0.2,10.8.0.3.10.8.0.4,10.8.0.5 -d 10.8.0.1 -j ACCEPT`
and drop other connections:
`iptables -A FORWARD -i wg0 -s 10.8.0.2,10.8.0.3.10.8.0.4,10.8.0.5 -d 10.8.0.0/24 -j DROP`

is it right?

**Reply by Pleonazm** [reply]
Yes, it looks good.
As always check how it works with your other IPtables stuff (like default policy). -A adds rules at the end, so if there is something allowing packets earlier, it will do it before these isolation rules
Does your client has something behind WG, that he accesses? In that case, you have to allow/block this IP as needed
One more thing. You also may add this. It allows packets destined directly to the server, not the ones routed somewhere else (like in the FORWARD chain).

iptables -A INPUT -i wg0 -s 10.8.0.2,10.8.0.3.10.8.0.4,10.8.0.5 -d 10.8.0.1 -j ACCEPT

**Comment by azat385** [comment]
According to [https://shibumi.dev/posts/isolated-clients-with-wireguard/](https://shibumi.dev/posts/isolated-clients-with-wireguard/)
Just add:
`iptables -I FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited`

After running:
`docker exec -it containerId iptables -I FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited`

Modified iptables of docker container:
```
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-A INPUT -p udp -m udp --dport 51820 -j ACCEPT
-A FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited
-A FORWARD -i wg0 -j ACCEPT
-A FORWARD -o wg0 -j ACCEPT

```

**Reply by azat385** [reply]
> docker exec -it containerId iptables -I FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited

**how to add that row while starting the container?**

should i add`-e WG_POST_UP=iptables -I FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited \ ` while staring the docker?

or modify `PostUp =.....` row  in `~/.wg-easy/wg0.conf` file?

**Reply by Pleonazm** [reply]
Heh, sometimes the simplest solutions are the best. I forgot about something like that
IIRC the wg0.conf file is replaced when anything is changed in configuration.
I suppose you should change src/config.js instead, then rebuild the docker image locally, and then use it in your compose file if you want to do it in this way.
Sending via environment variable overrides the rules (also it should be in quotes, like: -e WG_POST_UP="iptables -I FORWARD -i wg0 -o wg0 -j REJECT --reject-with icmp-admin-prohibited"), including NAT

**Comment by ghost** [comment]
For other that might see this issue : 
in my case I've added : -e WG_PRE_UP=iptables -I FORWARD -i wg0 -o wg0 -j DROP \ 

it filter server side (never trust the client) and it implement it before the tunnel is enabled.
